### PR TITLE
Migrate loggin to klogv2

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

File: controllers/nodefeaturediscovery_controls.go [hunk 1]
```diff
@@ -126,18 +125,18 @@ func ServiceAccount(n NFD) (ResourceStatus, error) {
 	// attempt to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating ")
+		klog.Info("ServiceAccount %q, Not found, creating", obj.Name)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:128]:
```suggestion
		klog.Info("ServiceAccount %q not found, creating", obj.Name)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 2]
```diff
@@ -155,19 +154,18 @@ func ClusterRole(n NFD) (ResourceStatus, error) {
...
-	logger.Info("Looking for")
+	klog.Info("Looking for ClusterRole %q on Namespace %", obj.Name, obj.Namespace)
 
 	// Look for the ClusterRole to see if it exists, and if so, check
 	// if it's Ready/NotReady. If the ClusterRole does not exist, then
 	// attempt to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("ClusterRole %q, Not found, creating", obj.Name)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:165]:
```suggestion
		klog.Info("ClusterRole %q not found, creating", obj.Name)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 3]
```diff
@@ -176,7 +174,7 @@ func ClusterRole(n NFD) (ResourceStatus, error) {
 	}
 
 	// If we found the ClusterRole, let's attempt to update it
-	logger.Info("Found, updating")
+	klog.Info("ClusterRole Found, updating", obj.Name)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:177]:
```suggestion
	klog.Info("ClusterRole found, updating", obj.Name)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 4]
```diff
@@ -223,7 +220,7 @@ func ClusterRoleBinding(n NFD) (ResourceStatus, error) {
 	}
 
 	// If we found the ClusterRoleBinding, let's attempt to update it
-	logger.Info("Found, updating")
+	klog.Info("ClusterRoleBinding %q Found on Namespace %q, updating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:223]:
```suggestion
	klog.Info("ClusterRoleBinding %q found in Namespace %q, updating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 5]
```diff
@@ -198,23 +196,22 @@ func ClusterRoleBinding(n NFD) (ResourceStatus, error) {
...
-	logger.Info("Looking for")
+	klog.Info("Looking for ClusterRoleBinding %q on Namespace %q", obj.Name, obj.Namespace)
 
 	// Look for the ClusterRoleBinding to see if it exists, and if so,
 	// check if it's Ready/NotReady. If the ClusterRoleBinding does not
 	// exist, then attempt to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("ClusterRoleBinding %q Not found on Namespace %q, creating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:211]:
```suggestion
		klog.Info("ClusterRoleBinding %q not found in Namespace %q, creating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 6]
```diff
@@ -264,10 +260,10 @@ func Role(n NFD) (ResourceStatus, error) {
 	// Ready/NotReady. If the Role does not exist, then attempt to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("Role %q, Not found on Namespace %q, creating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:263]:
```suggestion
		klog.Info("Role %q not found in Namespace %q, creating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 7]
```diff
@@ -318,10 +313,10 @@ func RoleBinding(n NFD) (ResourceStatus, error) {
 	// to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("RoleBinding %q, Not found on Namespace %q, creating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:316]:
```suggestion
		klog.Info("RoleBinding %q not found in Namespace %q, creating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 8]
```diff
@@ -330,7 +325,7 @@ func RoleBinding(n NFD) (ResourceStatus, error) {
 	}
 
 	// If we found the RoleBinding, let's attempt to update it
-	logger.Info("Found, updating")
+	klog.Info("RoleBinding %q Found on Namespace %q, updating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:328]:
```suggestion
	klog.Info("RoleBinding %q found in Namespace %q, updating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 9]
```diff
@@ -376,10 +370,10 @@ func ConfigMap(n NFD) (ResourceStatus, error) {
 	// it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("ConfigMap %q Not found on Namespace %q, creating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:373]:
```suggestion
		klog.Info("ConfigMap %q not found in Namespace %q, creating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 10]
```diff
@@ -462,9 +456,8 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 
 	// found states if the DaemonSet was found
 	found := &appsv1.DaemonSet{}
-	logger := log.WithValues("DaemonSet", obj.Name, "Namespace", obj.Namespace)
 
-	logger.Info("Looking for")
+	klog.Info("looking for Daemonset %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:460]:
```suggestion
	klog.Info("Looking for Daemonset %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 11]
```diff
@@ -479,10 +472,10 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 	// create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("Daemonset %q on Namespace %q Not found, creating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:475]:
```suggestion
		klog.Info("Daemonset %q in Namespace %q not found, creating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 12]
```diff
@@ -491,7 +484,7 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 	}
 
 	// If we found the DaemonSet, let's attempt to update it
-	logger.Info("Found, updating")
+	klog.Info("Daemonset %q on Namespace %q Found, updating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:487]:
```suggestion
	klog.Info("Daemonset %q in Namespace %q found, updating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 13]
```diff
@@ -528,9 +521,8 @@ func Service(n NFD) (ResourceStatus, error) {
 
 	// found states if the Service was found
 	found := &corev1.Service{}
-	logger := log.WithValues("Service", obj.Name, "Namespace", obj.Namespace)
 
-	logger.Info("Looking for")
+	klog.Info("looking for Service %q in Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:525]:
```suggestion
	klog.Info("Looking for Service %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 14]
```diff
@@ -545,18 +537,18 @@ func Service(n NFD) (ResourceStatus, error) {
 	// it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("Service %q in Namespace %q, Not found, creating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:540]:
```suggestion
		klog.Info("Service %q in Namespace %q not found, creating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 15]
```diff
@@ -198,23 +196,22 @@ func ClusterRoleBinding(n NFD) (ResourceStatus, error) {
...
 	// found states if the ClusterRoleBinding was found
 	found := &rbacv1.ClusterRoleBinding{}
-	logger := log.WithValues("ClusterRoleBinding", obj.Name, "Namespace", obj.Namespace)
 
 	// It is also assumed that our ClusterRoleBinding has a defined
 	// Namespace
 	obj.Subjects[0].Namespace = n.ins.GetNamespace()
 
-	logger.Info("Looking for")
+	klog.Info("Looking for ClusterRoleBinding %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:204]: supert nit: in/on namespace, use consistently 😄 

```suggestion
	klog.Info("Looking for ClusterRoleBinding %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 16]
```diff
@@ -198,23 +196,22 @@ func ClusterRoleBinding(n NFD) (ResourceStatus, error) {
...
 	// check if it's Ready/NotReady. If the ClusterRoleBinding does not
 	// exist, then attempt to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("ClusterRoleBinding %q Not found on Namespace %q, creating", obj.Name, obj.Namespace)
 		err = n.rec.Client.Create(context.TODO(), &obj)
 		if err != nil {
-			logger.Info("Couldn't create")
+			klog.Info("Couldn't create ClusterRoleBinding %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:214]:
```suggestion
			klog.Info("Couldn't create ClusterRoleBinding %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 17]
```diff
@@ -248,9 +245,8 @@ func Role(n NFD) (ResourceStatus, error) {
 
 	// found states if the Role was found
 	found := &rbacv1.Role{}
-	logger := log.WithValues("Role", obj.Name, "Namespace", obj.Namespace)
 
-	logger.Info("Looking for")
+	klog.Info("Looking for Role %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:249]:
```suggestion
	klog.Info("Looking for Role %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 18]
```diff
@@ -264,10 +260,10 @@ func Role(n NFD) (ResourceStatus, error) {
 	// Ready/NotReady. If the Role does not exist, then attempt to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("Role %q, Not found on Namespace %q, creating", obj.Name, obj.Namespace)
 		err = n.rec.Client.Create(context.TODO(), &obj)
 		if err != nil {
-			logger.Info("Couldn't create")
+			klog.Info("Couldn't create Role %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:266]:
```suggestion
			klog.Info("Couldn't create Role %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 19]
```diff
@@ -276,7 +272,7 @@ func Role(n NFD) (ResourceStatus, error) {
 	}
 
 	// If we found the Role, let's attempt to update it
-	logger.Info("Found, updating")
+	klog.Info("Found Role %q on Namespace %q, updating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:275]:
```suggestion
	klog.Info("Found Role %q in Namespace %q, updating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 20]
```diff
@@ -302,9 +298,8 @@ func RoleBinding(n NFD) (ResourceStatus, error) {
 
 	// found states if the RoleBinding was found
 	found := &rbacv1.RoleBinding{}
-	logger := log.WithValues("RoleBinding", obj.Name, "Namespace", obj.Namespace)
 
-	logger.Info("Looking for")
+	klog.Info("Looking for RoleBinding %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:302]:
```suggestion
	klog.Info("Looking for RoleBinding %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 21]
```diff
@@ -318,10 +313,10 @@ func RoleBinding(n NFD) (ResourceStatus, error) {
 	// to create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("RoleBinding %q, Not found on Namespace %q, creating", obj.Name, obj.Namespace)
 		err = n.rec.Client.Create(context.TODO(), &obj)
 		if err != nil {
-			logger.Info("Couldn't create")
+			klog.Info("Couldn't create RoleBinding %q on namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:319]:
```suggestion
			klog.Info("Couldn't create RoleBinding %q in namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 22]
```diff
@@ -359,9 +354,8 @@ func ConfigMap(n NFD) (ResourceStatus, error) {
 
 	// found states if the ConfigMap was found
 	found := &corev1.ConfigMap{}
-	logger := log.WithValues("ConfigMap", obj.Name, "Namespace", obj.Namespace)
 
-	logger.Info("Looking for")
+	klog.Info("Looking for ConfigMap %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:358]:
```suggestion
	klog.Info("Looking for ConfigMap %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 23]
```diff
@@ -376,10 +370,10 @@ func ConfigMap(n NFD) (ResourceStatus, error) {
 	// it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("ConfigMap %q Not found on Namespace %q, creating", obj.Name, obj.Namespace)
 		err = n.rec.Client.Create(context.TODO(), &obj)
 		if err != nil {
-			logger.Info("Couldn't create")
+			klog.Info("Couldn't create ConfigMap %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:376]:
```suggestion
			klog.Info("Couldn't create ConfigMap %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 24]
```diff
@@ -388,7 +382,7 @@ func ConfigMap(n NFD) (ResourceStatus, error) {
 	}
 
 	// If we found the ConfigMap, let's attempt to update it
-	logger.Info("Found, updating")
+	klog.Info("Found ConfigMap %q on Namespace %q, updating", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:385]:
```suggestion
	klog.Info("Found ConfigMap %q in Namespace %q, updating", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 25]
```diff
@@ -479,10 +472,10 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 	// create it
 	err := n.rec.Client.Get(context.TODO(), types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, found)
 	if err != nil && errors.IsNotFound(err) {
-		logger.Info("Not found, creating")
+		klog.Info("Daemonset %q on Namespace %q Not found, creating", obj.Name, obj.Namespace)
 		err = n.rec.Client.Create(context.TODO(), &obj)
 		if err != nil {
-			logger.Info("Couldn't create")
+			klog.Info("Couldn't create Daemonset %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:478]:
```suggestion
			klog.Info("Couldn't create Daemonset %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 26]
```diff
@@ -109,9 +109,8 @@ func ServiceAccount(n NFD) (ResourceStatus, error) {
 
 	// found states if the ServiceAccount was found
 	found := &corev1.ServiceAccount{}
-	logger := log.WithValues("ServiceAccount", obj.Name, "Namespace", obj.Namespace)
 
-	logger.Info("Looking for")
+	klog.Info("Looking for ServiceAccount %q on Namespace %q", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:113]: supert nit: in/on namespace, use consistently 😄 

```suggestion
	klog.Info("Looking for ServiceAccount %q in Namespace %q", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 27]
```diff
@@ -126,18 +125,18 @@ func ServiceAccount(n NFD) (ResourceStatus, error) {
...
+			klog.Info("Couldn't create ServiceAccount %q", obj.Name)
 			return NotReady, err
 		}
 		return Ready, nil
 	} else if err != nil {
 		return NotReady, err
 	}
 
-	logger.Info("Found, skipping update")
+	klog.Info("Found ServiceAccount %q on Namespace %q, skipping update", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:139]:
```suggestion
	klog.Info("Found ServiceAccount %q in Namespace %q, skipping update", obj.Name, obj.Namespace)
```

File: controllers/nodefeaturediscovery_controls.go [hunk 28]
```diff
@@ -155,19 +154,18 @@ func ClusterRole(n NFD) (ResourceStatus, error) {
 
 	// found states if the ClusterRole was found
 	found := &rbacv1.ClusterRole{}
-	logger := log.WithValues("ClusterRole", obj.Name, "Namespace", obj.Namespace)
 
-	logger.Info("Looking for")
+	klog.Info("Looking for ClusterRole %q on Namespace %", obj.Name, obj.Namespace)
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:158]:
```suggestion
	klog.Info("Looking for ClusterRole %q in Namespace %", obj.Name, obj.Namespace)
```

**Review by marquiz — COMMENTED** [review]
Thanks, looks good, now 👍 Just some nitting on the consistency of log messages, sorry 🙄  Will take a fresh look tomorrow with fresh eyes

**Review by marquiz — CHANGES_REQUESTED** [review]
Could you squash the two latter commits into one. And, start the commit message (of the second commit) with a capital letter 😉 

Otherwise lgtm 👍

**Comment by ArangoGutierrez** [comment]
> Could you squash the two latter commits into one. And, start the commit message (of the second commit) with a capital letter wink
> 
> Otherwise lgtm +1

Done

**Review by marquiz — APPROVED** [review]
Looks ready to rock mainline 🎸 
/lgtm

**Pull request merged**